dbms/src/Common/MemoryTracker.cpp

```diff
@@ -111,7 +116,7 @@ void MemoryTracker::alloc(Int64 size, bool check_memory_limit)
         Int64 current_bytes_rss_larger_than_limit = bytes_rss_larger_than_limit.load(std::memory_order_relaxed);
         bool is_rss_too_large = (!next.load(std::memory_order_relaxed) && current_limit
                                  && real_rss > current_limit + current_bytes_rss_larger_than_limit
-                                 && will_be > current_limit - (real_rss - current_limit - current_bytes_rss_larger_than_limit));
+                                 && will_be > baseline_of_query_mem_tracker);
```

Comment:
compare with address?

Reply:
type of `baseline_of_query_mem_tracker`  is `atomic<int64>`